src/Features/TransientNotices.php

```diff
@@ -0,0 +1,90 @@
+<?php
+/**
+ * WooCommerce Transient Notices
+ * NOTE: DO NOT edit this file in WooCommerce core, this is generated from woocommerce-admin.
+ */
+
+namespace Automattic\WooCommerce\Admin\Features;
+
+use \Automattic\WooCommerce\Admin\Loader;
+
+/**
+ * Shows print shipping label banner on edit order page.
+ */
+class TransientNotices {
+
+	/**
+	 * Option name for the queue.
+	 */
+	const QUEUE_OPTION = 'woocommerce_admin_transient_notices_queue';
+
+	/**
+	 * Constructor
+	 */
+	public function __construct() {
+		add_action( 'admin_enqueue_scripts', array( __CLASS__, 'enqueue_notices' ) );
+	}
+
+	/**
+	 * Add a notice to be shown.
+	 *
+	 * @param array $notice Notice.
+	 *    $notice = array(
+	 *      'id'      => (string) Unique ID for the notice. Required.
+	 *      'status'  => (string) info|error|success
+	 *      'content' => (string) Content to be shown for the notice. Required.
+	 *      'options' => (array) Array of options to be passed to the notice component.
+	 *       See https://developer.wordpress.org/block-editor/reference-guides/data/data-core-notices/#createNotice for available options.
+	 *    ).
+	 */
+	public static function add( $notice ) {
+		$queue = get_option( self::QUEUE_OPTION, array() );
+
+		$defaults               = array(
+			'status'  => 'info',
+			'options' => array(),
+		);
+		$notice_data            = array_merge( $defaults, $notice );
+		$notice_data['options'] = (object) $notice_data['options'];
+
+		$queue[ $notice['id'] ] = $notice_data;
+		update_option( self::QUEUE_OPTION, $queue );
+	}
+
+	/**
+	 * Remove a notice by ID.
+	 *
+	 * @param array $notice_id Notice of ID to remove.
+	 */
+	public static function remove( $notice_id ) {
+		$queue = get_option( self::QUEUE_OPTION );
+		unset( $queue[ $notice_id ] );
+		update_option( self::QUEUE_OPTION, $queue );
+	}
+
+	/**
+	 * Enqueue notices to be displayed on page load.
+	 */
+	public static function enqueue_notices() {
+		$notices = get_option( self::QUEUE_OPTION );
+
+		if ( empty( $notices ) ) {
+			return;
+		}
+
+		$script_assets_filename = Loader::get_script_asset_filename( 'transient-notices' );
+		$script_assets          = require WC_ADMIN_ABSPATH . WC_ADMIN_DIST_JS_FOLDER . 'wp-admin-scripts/' . $script_assets_filename;
+
+		wp_enqueue_script(
+			'transient-notices',
+			Loader::get_url( 'wp-admin-scripts/transient-notices', 'js' ),
+			array_merge( array( WC_ADMIN_APP ), $script_assets ['dependencies'] ),
+			WC_ADMIN_VERSION_NUMBER,
+			true
+		);
+
+		wp_localize_script( 'transient-notices', 'notices', array_values( $notices ) );
+		delete_option( self::QUEUE_OPTION );
```

Comment:
I think this has a pitfall, if say the user did not end up viewing the notice (like disconnected halfway through or navigated out just before the page finished loading), the notices will get deleted.

In CES we clear the queue via a trigger in `updateOptions` in @wordpress/data AFAIK, I think we can implement a similar thing here

Reply:
I saw this and thought that may be the reason for updating this client-side.  I can see an error occurring from another plugin which would then prevent the notice from ever being shown.

I think this is worthwhile to add to this PR.